### PR TITLE
[dev-v2.7] longhorn 102.4.0+up1.6.1 add

### PR DESCRIPTION
**Pull request by mantissahz**
Release Longhorn v1.6.1 into Rancher 2.7

Longhorn https://github.com/longhorn/longhorn/issues/8209
Rancher https://github.com/rancher/rancher/issues/44861

Signed-off-by: James Lu [james.lu@suse.com](mailto:james.lu@suse.com)

cc @innobead @PhanLe1010 @khushboo-rancher @roger-ryao

Waiting for Longhorn 1.6.1 images merged into the Rancher image-mirror list.

### CODE REVIEW

**Review by nicholasSUSE — APPROVED** [review]
Hello, we have improved the Charts documentation regarding Pull Request Rules.

Please take a look at it before making any new Pull Requests.
We are still approving Pull Requests and starting to inform all teams, soon we will require the standard in the documentation.

https://github.com/rancher/charts?tab=readme-ov-file#pull-request-rules

File: packages/longhorn-crd/longhorn-1.6/package.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+url: https://github.com/longhorn/charts.git
+subdirectory: charts/longhorn
+commit: 35833202fc0df11d6ee61b735990a8d0356b29e4
+version: 102.4.1
```
Comment (PhanLe1010) [inline packages/longhorn-crd/longhorn-1.6/package.yaml]: This is the first 1.6.x version that we released in Rancher charts. I think it should be `102.4.0` instead. WDYT?

File: packages/longhorn/longhorn-1.6/package.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+url: https://github.com/longhorn/charts.git
+subdirectory: charts/longhorn
+commit: 35833202fc0df11d6ee61b735990a8d0356b29e4
+version: 102.4.1
```
Comment (PhanLe1010) [inline packages/longhorn/longhorn-1.6/package.yaml]: Same as above: This is the first 1.6.x version that we released in Rancher charts. I think it should be `102.4.0` instead. WDYT?

File: packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/validate-install-crd.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#{{- if gt (len (lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "")) 0 -}}
+# {{- $found := dict -}}
+# {{- set $found "longhorn.io/v1beta1/BackingImageDataSource" false -}}
+# {{- set $found "longhorn.io/v1beta1/BackingImageManager" false -}}
+# {{- set $found "longhorn.io/v1beta1/BackingImage" false -}}
```
Comment (PhanLe1010) [inline packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/validate-install-crd.yaml:5]: We are missing the CRD `backupbackingimages.longhorn.io`

File: packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/userroles.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+rules:
+- apiGroups: [ "longhorn.io" ]
+  resources: ["volumes", "volumes/status", "engines", "engines/status", "replicas", "replicas/status", "settings",
+              "engineimages", "engineimages/status", "nodes", "nodes/status", "instancemanagers", "instancemanagers/status",
+              "sharemanagers", "sharemanagers/status", "backingimages", "backingimages/status",
+              "backingimagemanagers", "backingimagemanagers/status", "backingimagedatasources", "backingimagedatasources/status",
+              "backuptargets", "backuptargets/status", "backupvolumes", "backupvolumes/status", "backups", "backups/status",
+              "recurringjobs", "recurringjobs/status", "orphans", "orphans/status", "snapshots", "snapshots/status",
+              "supportbundles", "supportbundles/status", "systembackups", "systembackups/status", "systemrestores", "systemrestores/status",
+              "volumeattachments", "volumeattachments/status"]
```
Comment (PhanLe1010) [inline packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/userroles.yaml:34]: We are missing the CRD `backupbackingimages.longhorn.io`

File: charts/longhorn/102.4.1+up1.6.1/templates/validate-install-crd.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#{{- if gt (len (lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "")) 0 -}}
+# {{- $found := dict -}}
+# {{- set $found "longhorn.io/v1beta1/BackingImageDataSource" false -}}
+# {{- set $found "longhorn.io/v1beta1/BackingImageManager" false -}}
+# {{- set $found "longhorn.io/v1beta1/BackingImage" false -}}
```
Comment (PhanLe1010) [inline charts/longhorn/102.4.1+up1.6.1/templates/validate-install-crd.yaml]: We are missing the CRD `backupbackingimages.longhorn.io`

File: charts/longhorn/102.4.1+up1.6.1/templates/userroles.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+rules:
+- apiGroups: [ "longhorn.io" ]
+  resources: ["volumes", "volumes/status", "engines", "engines/status", "replicas", "replicas/status", "settings",
+              "engineimages", "engineimages/status", "nodes", "nodes/status", "instancemanagers", "instancemanagers/status",
+              "sharemanagers", "sharemanagers/status", "backingimages", "backingimages/status",
+              "backingimagemanagers", "backingimagemanagers/status", "backingimagedatasources", "backingimagedatasources/status",
+              "backuptargets", "backuptargets/status", "backupvolumes", "backupvolumes/status", "backups", "backups/status",
+              "recurringjobs", "recurringjobs/status", "orphans", "orphans/status", "snapshots", "snapshots/status",
+              "supportbundles", "supportbundles/status", "systembackups", "systembackups/status", "systemrestores", "systemrestores/status",
+              "volumeattachments", "volumeattachments/status"]
```
Comment (PhanLe1010) [inline charts/longhorn/102.4.1+up1.6.1/templates/userroles.yaml]: We are missing the CRD `backupbackingimages.longhorn.io`

File: packages/longhorn-crd/longhorn-1.6/package.yaml (hunk 1 above)
Comment (mantissahz) [reply packages/longhorn-crd/longhorn-1.6/package.yaml]: Modified. Thanks

File: packages/longhorn/longhorn-1.6/package.yaml (hunk 2 above)
Comment (mantissahz) [reply packages/longhorn/longhorn-1.6/package.yaml]: Modified. Thanks

File: packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/validate-install-crd.yaml (hunk 3 above)
Comment (mantissahz) [reply packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/validate-install-crd.yaml:5]: Added. Thanks

File: charts/longhorn/102.4.1+up1.6.1/templates/validate-install-crd.yaml (hunk 5 above)
Comment (mantissahz) [reply charts/longhorn/102.4.1+up1.6.1/templates/validate-install-crd.yaml]: Added. Thanks

File: packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/userroles.yaml (hunk 4 above)
Comment (mantissahz) [reply packages/longhorn/longhorn-1.6/generated-changes/overlay/templates/userroles.yaml:34]: Added. Thanks

File: charts/longhorn/102.4.1+up1.6.1/templates/userroles.yaml (hunk 6 above)
Comment (mantissahz) [reply charts/longhorn/102.4.1+up1.6.1/templates/userroles.yaml]: Added. Thanks

**Comment by roger-ryao** [comment]
Prerequisites:
- Set the Concurrent Automatic Engine Upgrade Per Node Limit to a value greater than 0.
- `LONGHORN_REPO` is set to `longhornio`.
- Rancher : `v2.7.12`

Verify is passed as follows step
- [x] Fresh installation
- [x] Uninstallation
- [x] Upgrade from `102.3.2+up1.5.4` - > `102.4.0+up1.6.1`

**Review by PhanLe1010 — APPROVED** [review]
LGTM

**Review by PhanLe1010 — APPROVED** [review]

**Comment by PhanLe1010** [comment]
Cannot merge because the image validation script failed with error:
```
time="2024-04-09T20:08:58Z" level=fatal msg="failed to generate namespace and repository for image: quay.io/openshift/origin-oauth-proxy"
make: *** [Makefile:17: check-images] Error 1
Error: Process completed with exit code 2.
```
It is is failed at this line of code https://github.com/rancher/charts-build-scripts/blob/a7001e331fda7b61c8df358862f46436eaae9df2/pkg/images/checkImages.go#L55-L59. It means that the recent added image in Longhorn, `quay.io/openshift/origin-oauth-proxy`, doesn't follow the format `<repo>/<image-name>`.

I think even if we solve the image format, the script will still failed because `quay.io/openshift/origin-oauth-proxy` is not mirrored to Longhorn repo and to Rancher repo yet. We should mirror them similar to what we did for CSI component. 

Creating a new GitHub ticket: https://github.com/longhorn/longhorn/issues/8329

**Comment by PhanLe1010** [comment]
Waiting for the image PR (adding missing image) https://github.com/rancher/image-mirror/pull/608

File: charts/longhorn/102.4.0+up1.6.1/values.yaml [hunk 7]
```diff
@@ -0,0 +1,483 @@
...
+      tag: v6.3.2
+    livenessProbe:
+      # -- Repository for the CSI liveness probe image. When unspecified, Longhorn uses the default value.
+      repository: rancher/mirrored-longhornio-livenessprobe
+      # -- Tag for the CSI liveness probe image. When unspecified, Longhorn uses the default value.
+      tag: v2.12.0
+  openshift:
+    oauthProxy:
+      # -- Repository for the OAuth Proxy image. This setting applies only to OpenShift users.
+      repository: rancher/mirrored-longhornio-openshift-origin-oauth-proxy
```
Comment (PhanLe1010) [inline charts/longhorn/102.4.0+up1.6.1/values.yaml:103]: Hi @mantissahz we need to add this mirror image to questions.yaml too. Could you help to update the PR?

**Review by PhanLe1010 — CHANGES_REQUESTED** [review]

File: charts/longhorn/102.4.0+up1.6.1/values.yaml (hunk 7 above)
Comment (mantissahz) [reply charts/longhorn/102.4.0+up1.6.1/values.yaml:103]: Added. Thanks

**Review by PhanLe1010 — APPROVED** [review]
LGTM

**Comment by PhanLe1010** [comment]
@khushboo-rancher Because we have had significant modifications since the last test run, could we retest the PR? Once it pass, I will merge the PR

**Comment by khushboo-rancher** [comment]
QA - PASS

Validated below scenarios:
1. Installation
2. uninstallation
3. upgrade
4. P0 test cases - mounting volume (RWO, RWX), replica rebuilding, backing image etc

**Pull request merged**